query.go

```diff
@@ -662,21 +662,31 @@ checkResponses:
 			}
 
 			queryPeer = nil
-			for _, curPeer := range s.Peers() {
-				if curPeer != nil && curPeer.Connected() &&
-					peerTries[curPeer.Addr()] < qo.numRetries {
-
-					curPeer := curPeer
-					queryPeer = curPeer
-
-					// Found a peer we can query.
-					peerTries[queryPeer.Addr()]++
-					queryPeer.subscribeRecvMsg(subscription)
-					queryPeer.QueueMessageWithEncoding(
-						queryMsg, nil, qo.encoding,
-					)
-					break
+			for _, peer := range s.Peers() {
+				// If the peer is no longer connected, we'll
+				// skip them.
+				if !peer.Connected() {
```

Comment:
Missing `peer != nil` check

Reply:
Why is the check needed?

Reply:
To prevent panics, check the history

Reply:
Are you referring to https://github.com/lightninglabs/neutrino/commit/9a0709b914d3ea43113806d8b8aaf6f87e06bf44? That seems different -- we query for a peer given their address. The `nil` check makes sense there to me as the peer can no longer exist, but here `Peers` is just returning the currently active peers.

Reply:
Hmm, actually i think I’m recalling a different case where checking nilness is necessary. Why is used atm?

Reply:
I think it was left over from some shadowing changes, see https://github.com/lightninglabs/neutrino/commit/fd24569c15b5960f51dc3f336954b8406a2beac1.

Reply:
Ahh okay, thanks for clarifying. Clear to me now that it is not needed